manager/orchestrator/update/updater.go

```diff
@@ -521,7 +521,11 @@ func (u *Updater) removeOldTasks(ctx context.Context, batch *store.Batch, remove
 }
 
 func (u *Updater) isTaskDirty(t *api.Task) bool {
-	return orchestrator.IsTaskDirty(u.newService, t)
+	var n *api.Node
+	u.store.View(func(tx store.ReadTx) {
+		n = store.GetNode(tx, t.NodeID)
+	})
+	return orchestrator.IsTaskDirty(u.newService, t, n)
```

Comment:
What happens if the node spec changes between the read and the check? Would it be too heavy, or cause locking conflicts, if we checked the constraints inside of the read tx?

Reply:
This is a legit concern, but I think it's reasonable to expect users to not make frequent consecutive updates to the node and service specs.

Reply:
Oh just saw @aaronlehmann's comment, never mind.